### PR TITLE
UI Task Creation + Split UI up into components

### PR DESCRIPTION
**Pull request by zontasticality**

### CODE REVIEW

File: client/src/main.rs [hunk 1]
```diff
@@ -9,39 +9,32 @@ use std::{
...
 mod mid;
 mod term;
-
-const BACKGROUND: Color = Color::Reset;
-const TEXT_COLOR: Color = Color::White;
-const SELECTED_STYLE_FG: Color = Color::LightYellow;
-const COMPLETED_TEXT_COLOR: Color = Color::Green;
+mod ui;
 
 #[coverage(off)]
```
Comment (knoxy5467) [inline client/src/main.rs:21]: we talked about this coverage so thats ok

File: client/src/main.rs [hunk 2]
```diff
@@ -70,361 +63,23 @@ pub fn install_hooks() -> color_eyre::Result<()> {
 
     // used color_eyre's PanicHook as the standard panic hook
     let panic_hook = panic_hook.into_panic_hook();
-    panic::set_hook(Box::new(#[coverage(off)] move |panic_info| {
-        term::restore().unwrap();
-        panic_hook(panic_info);
-    }));
+    panic::set_hook(Box::new(
+        #[coverage(off)]
```
Comment (knoxy5467) [inline client/src/main.rs:67]: why is this coverage off?

File: client/src/main.rs [hunk 3]
```diff
@@ -70,361 +63,23 @@ pub fn install_hooks() -> color_eyre::Result<()> {
...
+    ));
 
     // use color_eyre's EyreHook as eyre's ErrorHook
     let eyre_hook = eyre_hook.into_eyre_hook();
-    eyre::set_hook(Box::new(#[coverage(off)] move |error| {
-        term::restore().unwrap();
-        eyre_hook(error)
-    }))?;
+    eyre::set_hook(Box::new(
+        #[coverage(off)]
```
Comment (knoxy5467) [inline client/src/main.rs:77]: why is this coverage off?

**Review by knoxy5467 — CHANGES_REQUESTED** [review]
There is a lot of coverage being ignored without explination. if coverage is going to be ignored then you need to explain why it needs to be ignored and why it cannot be tested. we already talked about the main function but these other functions have no attached reasons

File: client/src/main.rs (hunk 2 above)
Comment (zontasticality) [reply client/src/main.rs]: `#[coverage(off)]` is not recursive, the top one doesn't coverage closures so you need to annotate every closure.

**Comment by zontasticality** [comment]
Old PR

**Pull request closed without merging**